providers/form/validate.go

```diff
@@ -0,0 +1,21 @@
+package form
+
+import "github.com/asaskevich/govalidator"
+
+type Complexity struct {
+	MinLength int
+	MaxLength int
+	Patterns  []string
```

Comment:
I thought an array of patterns would allow you to specify multiple criteria, e.g. uppercase/lowercase/numeric etc.  My regex is pretty good but I couldn't think how to do that in a single regex.